test/smoke/src/sql/areas/notebook/notebook.test.ts

```diff
@@ -94,6 +94,33 @@ export function setup() {
 			const app = this.app as Application;
 			await app.workbench.quickaccess.runCommand('workbench.action.revertAndCloseActiveEditor');
 		});
+
+		describe('Notebook Toolbar Actions', async () => {
+
+			it('Add Code Cell', async function () {
+				const app = this.app as Application;
+				await app.workbench.sqlNotebook.newUntitledNotebook();
+				await app.workbench.sqlNotebook.addCell('code');
+				await app.workbench.sqlNotebook.waitForTypeInEditor('Select * FROM Customers;');
+			});
+
+			it('Add Markdown Cell', async function () {
+				const app = this.app as Application;
+				await app.workbench.sqlNotebook.newUntitledNotebook();
+				await app.workbench.sqlNotebook.addCellFromPlaceholder('Markdown');
+				await app.workbench.sqlNotebook.waitForPlaceholderGone();
+			});
+
+			it('Collapse and Expand Cell', async function () {
+				const app = this.app as Application;
+				await app.workbench.sqlNotebook.openFile('collapsed.ipynb');
+				await app.workbench.sqlNotebook.waitForCollapseCellsIcon();
```

Comment:
It might be easier to call `app.workbench.sqlNotebook.notebookToolbar.waitForCollapseCellsIcon()` here and then remove the methods from the Notebook class. Similar to how Cory added the textCellToolbar functionality here https://github.com/microsoft/azuredatastudio/pull/15868/files. I can do the clean up for the trust icon.

Reply:
Considering adding some sort of logic to detect whether the cell is actually collapsed/expanded - simplest would be to check for the collapse/expand button on the cell itself but if you could actually check the displayed text to verify that it's cut off/shown appropriately that would be the best.